code/game/objects/items/cards_ids.dm

```diff
@@ -421,16 +421,17 @@ update_label()
 					alert ("Invalid name.")
 					return
 			else
-				registered_name = t
+				registered_name = target_name
 
-			var/u = copytext(sanitize(input(user, "What occupation would you like to put on this card?\nNote: This will not grant any access levels other than Maintenance.", "Agent card job assignment", assignment ? assignment : "Assistant") as text | null),1,MAX_MESSAGE_LEN)
-			if(!u)
+			var/target_occupation = copytext(sanitize(input(user, "What occupation would you like to put on this card?\nNote: This will not grant any access levels other than Maintenance.", "Agent card job assignment", assignment ? assignment : "Assistant") as text | null),1,MAX_MESSAGE_LEN)
+			if(!target_occupation)
 				registered_name = ""
 				return
-			assignment = u
+			assignment = target_occupation
 			update_label()
 			forged = TRUE
 			to_chat(user, "<span class='notice'>You successfully forge the ID card.</span>")
+			log_game("[key_name(user)] has forged [src] with name \"[registered_name]\" and occupation \"[assignment]\".")

```

Comment:
Name changes based on registered_name and assignment, so [src] is the new card name here.

Reply:
Same here